libs/sparkpost-lib/src/main/java/com/sparkpost/transport/RestConnection.java

```diff
@@ -79,20 +73,22 @@ public RestConnection(Client client) throws SparkPostException {
      *
      * @param client
      *            Client object to use (in particular for authentication info)
-     * @param endpoint
+     * @param baseUrl
      *            Endpoint to use instead of the default defaultApiEndpoint
      * @throws SparkPostException
      */
-    public RestConnection(Client client, String endpoint) throws SparkPostException {
+    public RestConnection(Client client, String baseUrl) throws SparkPostException {
+
         this.client = client;
-        if (StringUtils.isAnyEmpty(endpoint)) {
-            this.endpoint = defaultApiEndpoint;
+        if (StringUtils.isAnyEmpty(baseUrl)) {
+            this.baseUrl = defaultApiEndpoint;
         } else {
-            if (endpoint.endsWith("/")) {
-                this.endpoint = endpoint;
-            } else {
-                this.endpoint = endpoint + '/';
-            }
+
+            this.baseUrl = baseUrl;
+        }
+
+        if (baseUrl.endsWith("/")) {
+            throw new IllegalStateException("SPARKPOST_BASE_URL should not end with a '/',  SPARKPOST_BASE_URL=" + baseUrl + "");
```

Comment:
It is a good question and I went back and forth on that very thing. This `throw` is really there to catch user configuration error. My assumption is the end user of the lib will see the error and easily fix it in config.properties. In most cases in SPC they will never even set the property this is primarily there for SPE and Momentum users.

I throwing an exception was better than doing even more string manipulation. So that is the background on why it is the way that it is.

To be honest I can go either way on it. If after reading this you feel it is better to remove the trailing slash I would be glad to change it to do that.